Inc/modem/modem.hpp

```diff
@@ -75,6 +75,14 @@ Q_OBJECT
 
     bool message(const std::string &number, const std::string &message);
 
+    bool setMessageMode(bool mode);
+
+    bool setNumberID(bool mode);
+
+    bool setEchoMode(bool mode);
```

Comment:
Consider marking these methods with [[ nodiscard ]] since the return value of these functions is always important for the CLI to determine whether the execution was successful.

```suggestion
    [[nodiscard]] bool setMessageMode(bool mode);

    [[nodiscard]] bool setNumberID(bool mode);

    [[nodiscard]] bool setEchoMode(bool mode);
```